crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -99,6 +107,22 @@ impl Context {
                 assert_eq!(result_ids.len(), 1, "Binary ops have a single result");
                 self.ssa_value_to_acir_var.insert(result_ids[0], result_acir_var);
             }
+            Instruction::Call { func, arguments: _ } => {
+                match &dfg[*func] {
+                    Value::Function(id) => {
+                        let func = &ssa.functions[id];
+                        match func.runtime() {
+                            RuntimeType::Normal => unreachable!(),
+                            RuntimeType::Unsafe => {
+                                // Generate the brillig code of the function
+                                let code = Artefact::default().link(func.dfg.obj());
```

Comment:
Why create a default object then link it with another instead of just using the other directly?

Reply:
Because the object is mutable and generate its own bytecode while the other is not modified.

Reply:
What is preventing us from doing `func.dfg.obj().clone()`?

Reply:
Using a new artefact makes it clearer that we start from a new one, although for now it does not really makes a difference, but link will do more than cloning.